baseapp/abci_utils.go

```diff
@@ -19,26 +19,34 @@ import (
 	"github.com/cosmos/cosmos-sdk/types/mempool"
 )
 
+// VoteExtensionThreshold defines the total voting power % that must be
+// submitted in order for all vote extensions to be considered valid for a
+// given height.
+var VoteExtensionThreshold = math.LegacyNewDecWithPrec(667, 3)
```

Comment:
something that came to me while reading this was what if an application says only these 5 validators need to do this vote extension and these other 10 have to do some other work, would that make sense to try or does that defeat the purpose a little?

Reply:
If they'd wanna do something like that, then I would argue they should implement their own `ValidateVoteExtensions` function. Recall, `ValidateVoteExtensions` is just a helper function we provide for apps. They're not forced to use it. They can also use it as a reference too if they wish.